### PR TITLE
bytes.mod implement with CFormat

### PR DESCRIPTION
**Pull request by yanganto**
implement byte formatting with bytes or tuple of bytes #736

### CODE REVIEW

**Comment by yanganto** [comment]
This PR is related to #1569

**Comment by yanganto** [comment]
working on `%i` and `%f`

**Comment by coolreader18** [comment]
I don't want to discourage you with all the work you've done so far, but printf-style bytes formatting [is pretty feature-equivalent](https://docs.python.org/3/library/stdtypes.html#printf-style-bytes-formatting) to the normal `str` version, e.g. it has named replacements, number formatting options, etc; so you might want to look in to modifying the existing code in `cformat.rs` to accept bytes.

**Comment by yanganto** [comment]
@coolreader18 Thanks for point out.  A suitable implementation should use the existing functions properly.  I will figure out how `cformat.rs` works in `str`.

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -1387,11 +1389,11 @@ fn try_update_quantity_from_tuple(
     }
 }
 
-fn do_cformat(
+pub fn do_cformat_string(
     vm: &VirtualMachine,
     mut format_string: CFormatString,
     values_obj: PyObjectRef,
-) -> PyResult {
+) -> Result<String, PyObjectRef> {
```
Comment (youknowone) [inline vm/src/obj/objstr.rs]: This is same to `PyResult<String>`

Comment (yanganto) [reply vm/src/obj/objstr.rs]: Thanks for the review.

**Comment by yanganto** [comment]
Thanks for your time.  Thanks for the review.

**Comment by coolreader18** [comment]
No problem, sorry it took this long.

**Pull request merged**